packages/datadog-instrumentations/test/mysql2.spec.js

```diff
@@ -435,16 +436,18 @@ describe('mysql2 instrumentation', () => {
               query.on('end', () => done())
             })
 
-            it('should work without abortController.abort()', (done) => {
+            it('should work without abortController.abort()', async () => {
               startCh.subscribe(noop)
               const query = pool.query(sql)
 
-              query.on('error', err => done(err))
-              query.on('end', () => {
-                sinon.assert.called(apmQueryStart)
+              let error
+              query.on('error', err => { error = err })
+              expect(query.listenerCount('error')).to.equal(1)
 
-                done()
-              })
+              await once(query, 'end')
+              expect(query.listenerCount('error')).to.equal(1)
+              sinon.assert.called(apmQueryStart)
+              expect(error).to.be.undefined
```

Comment:
This check, and the `error` listener you add above shouldn't be necessary, as the `await once(query, 'end')` will be rejected if `query` emits and error, and hence the test will fail and never reach this point.

Reply:
True, while it likely depends on the Node.js version to prevent an error from being emitted after 'end'. I'll simplify it.